dss/storage/validator.py

```diff
@@ -0,0 +1,78 @@
+"""Utilities in this file are used to removed extra fields from index data before adding to the index."""
+import re
+from jsonschema.validators import create, RefResolver
+from jsonschema import _utils, _validators
+
+import json
+from urllib.request import urlopen
+from urllib.error import URLError
+import ssl
+from typing import List
+import logging
+
+
+def read_json_url(link):
+    gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    f = urlopen(link, context=gcontext)
+    return json.load(f)
+
+
+def extract_json_fields(json_data: dict, path: List[str], fields: List[str]):
+    """
+        json_data: The json data from which to remove the field
+        path: The path through the json data to find the field.
```

Comment:
`A list of indices (either field names or array indices) forming a path through the JSON data.`

BTW: Have you thought of the possibility that a path element is an array index, rather than a dictionary key?

Reply:
Will change.
> BTW: Have you thought of the possibility that a path element is an array index, rather than a dictionary 
I did not initially, but that should not be a problem to navigate if it's in the path or remove if it's the field.

> key?

>